src/apps/Agreement/prop-types.js

```diff
@@ -0,0 +1,26 @@
+import PropTypes from 'prop-types'
+
+const AppItemType = PropTypes.shape({
+  entryActions: PropTypes.arrayOf(PropTypes.array),
+  allowedActions: PropTypes.arrayOf(PropTypes.string).isRequired,
+  actionCollateral: PropTypes.shape({
+    amount: PropTypes.number,
+    symbol: PropTypes.string,
+    address: PropTypes.string,
+  }),
+  challengeCollateral: PropTypes.shape({
+    amount: PropTypes.number,
+    symbol: PropTypes.string,
+    address: PropTypes.string,
+  }),
+  signerEligibility: PropTypes.shape({
+    amount: PropTypes.number,
+    symbol: PropTypes.string,
+    address: PropTypes.string,
+  }),
+  challengeEligibility: PropTypes.string.isRequired,
```

Comment:
This is something we can do later, but for context now:

`allowedActions`, `signerEligibility`, and `challengeEligibility` are concepts that will come from permissions.

`entryActions` may be similar, but it might actually be static (I think we'll only have "edit" as an item?).

Depending on how we build this prop (should we do a transformation step to reshape it [as we do with permissions data](https://github.com/aragon/aragon/blob/agreement__route-and-details/src/contexts/PermissionsContext.js)?), we may not keep these keys on this data type.

Reply:
I'm all ears on this one, I just made it align with the field names as a starting point. Let's address and update when the data goes in.